modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/EnterpriseGeoIpDownloaderTaskExecutor.java

```diff
@@ -0,0 +1,248 @@
+    }
+
+    @Override
+    public void clusterChanged(ClusterChangedEvent event) {
+        EnterpriseGeoIpDownloader currentDownloader = getCurrentTask();
+        if (currentDownloader != null) {
+            boolean hasGeoIpMetadataChanges = event.metadataChanged()
+                && event.changedCustomMetadataSet().contains(IngestGeoIpMetadata.TYPE);
+            if (hasGeoIpMetadataChanges) {
+                currentDownloader.requestReschedule(); // watching the cluster changed events to kick the thing off if it's not running
+            }
+        }
+    }
+
+    public synchronized void reload(Settings settings) {
+        // `SecureSettings` are available here! cache them as they will be needed
+        // whenever dynamic cluster settings change and we have to rebuild the accounts
+        try {
+            this.cachedSecureSettings = extractSecureSettings(settings, List.of(MAXMIND_LICENSE_KEY_SETTING));
+        } catch (GeneralSecurityException e) {
+            logger.error("Keystore exception while reloading enterprise geoip download task executor", e);
+        }
+    }
+
+    /**
+     * Extracts the {@link SecureSettings}` out of the passed in {@link Settings} object. The {@code Setting} argument has to have the
+     * {@code SecureSettings} open/available. Normally {@code SecureSettings} are available only under specific callstacks (eg. during node
+     * initialization or during a `reload` call). The returned copy can be reused freely as it will never be closed (this is a bit of
+     * cheating, but it is necessary in this specific circumstance). Only works for secure settings of type string (not file).
+     *
+     * @param source               A {@code Settings} object with its {@code SecureSettings} open/available.
+     * @param securePluginSettings The list of settings to copy.
+     * @return A copy of the {@code SecureSettings} of the passed in {@code Settings} argument.
+     */
+    private static SecureSettings extractSecureSettings(Settings source, List<Setting<?>> securePluginSettings)
+        throws GeneralSecurityException {
+        // get the secure settings out
+        final SecureSettings sourceSecureSettings = Settings.builder().put(source, true).getSecureSettings();
+        // filter and cache them...
+        final Map<String, Tuple<SecureString, byte[]>> cache = new HashMap<>();
+        if (sourceSecureSettings != null && securePluginSettings != null) {
+            for (final String settingKey : sourceSecureSettings.getSettingNames()) {
+                for (final Setting<?> secureSetting : securePluginSettings) {
+                    if (secureSetting.match(settingKey)) {
+                        cache.put(
+                            settingKey,
+                            new Tuple<>(sourceSecureSettings.getString(settingKey), sourceSecureSettings.getSHA256Digest(settingKey))
+                        );
+                    }
+                }
+            }
+        }
```

Comment:
This part I'm not familiar at all with, maybe someone like @jakelandis should take a look here and give it a sign off?

Reply:
That's a good suggestion -- I've hallway reviewed the code with him already, but I'll get an explicit +1 from him on this bit once we're doing with the rest.

Reply:
only reviewed the secure settings bits, LGTM pending the outcome of the error handling (see comment above)